pkg/operator2/route.go

```diff
@@ -15,7 +16,21 @@ import (
 )
 
 func (c *authOperator) handleRoute(ingress *configv1.Ingress) (*routev1.Route, *corev1.Secret, error) {
+	var updateRoute bool
 	route, err := c.route.Get(targetName, metav1.GetOptions{})
+	if err == nil {
+		// route exists, check to see if the ingress domain has changed
+		for _, ig := range route.Status.Ingress {
+			// we're only interested in the targetName. ingress
+			if !strings.HasPrefix(ig.Host, targetName) {
+				continue
+			}
+			if !strings.HasSuffix(ig.Host, "." + ingress.Spec.Domain) {
+				// the domain has changed, so update the route
+				updateRoute = true
+			}
+		}
+	}
```

Comment:
this is dangerous style to have if clauses without else ligned up. If you fall through with a modified err you get some funny behaviour.